pkg/controller/signup_test.go

```diff
@@ -644,29 +586,11 @@ func (s *TestSignupSuite) TestVerifyPhoneCodeHandler() {
 
 	s.Run("usersignup stored by its username", func() {
 		// Create another UserSignup
-		otherUserID := uuid.Must(uuid.NewV4()).String()
-
-		otherUserSignup := &crtapi.UserSignup{
-			TypeMeta: v1.TypeMeta{},
-			ObjectMeta: v1.ObjectMeta{
-				Name:      "jsmith",
-				Namespace: configuration.Namespace(),
-				Annotations: map[string]string{
-					crtapi.UserVerificationAttemptsAnnotationKey:   "0",
-					crtapi.UserSignupVerificationCodeAnnotationKey: "999127",
-					crtapi.UserVerificationExpiryAnnotationKey:     time.Now().Add(10 * time.Second).Format(service.TimestampLayout),
-				},
-			},
-			Spec: crtapi.UserSignupSpec{
-				IdentityClaims: crtapi.IdentityClaimsEmbedded{
-					PreferredUsername: "jsmith",
-					PropagatedClaims: crtapi.PropagatedClaims{
-						Sub: otherUserID,
-					},
-				},
-			},
-			Status: crtapi.UserSignupStatus{},
-		}
+		otherUserSignup := testusersignup.NewUserSignup(
```

Comment:
minor and not strictly related to your changes - it looks like now there is not much difference in how the "usersignup it's stored" compared to the other tests. I mean I'm not sure what the test case description refers to. I thought it was about how the `otherUserSignup` is configured ( with the `PropagatedClaims` filed being populated ) .

Reply:
Yes, you are completely right. It's a leftover that wasn't properly cleaned up after migration of UserSignups from using userIds to using usernames as the name of the resource. The end-goal is to clean this up, but this will require a lot of changes, so I'm doing it in small batches (for now some changes in unit tests that will help me to minimize the size of the changes of the final PRs). 
As for this test. It's not about how it's stored, but how it is retrieved - this test doesn't use the userID. Anyway, this sub-test will go away in the upcoming PR.